src/lib.rs

```diff
@@ -503,7 +512,8 @@ pub mod pallet {
 			T::Currency::unreserve(&who, reserve);
 
 			NonceCall::<T>::remove(supersig_index);
-			Supersigs::<T>::remove(supersig_index);
+			Members::<T>::remove_prefix(supersig_index, None);
+			TotalMembers::<T>::remove(supersig_index);
 			Calls::<T>::remove_prefix(supersig_index, None);
```

Comment:
Here we don't take into account all the money reserved for other calls preimage, that may be waiting for approval.

I suggest we loop over every call stored and unreserve the deposits associated with each preimage before we delete the supersig. Otherwise, the people will just lose their money when we clean up the storage

Reply:
Make a test that ensure this happen